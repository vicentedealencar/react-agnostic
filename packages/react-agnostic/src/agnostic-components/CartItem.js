import React from 'react'
import numeral from 'numeral'
import withComponents from '../withComponents'
import _UpdateCartItem from './UpdateCartItem'

const CartItem = ({
  item,
  components: { Text, View, Button },
  updateCartItem,
  UpdateCartItem = _UpdateCartItem,
  ...otherProps
}) => {
  const price = numeral(item.price)
    .multiply(item.amount)
    .format()

  return (
    <View {...otherProps}>
      <UpdateCartItem
        item={item}
        updateCartItem={updateCartItem}
        Button={Button}
        Text={Text}
      />
      <Text>
        {item.name} - {price}
      </Text>
    </View>
  )
}

export default withComponents(CartItem)
