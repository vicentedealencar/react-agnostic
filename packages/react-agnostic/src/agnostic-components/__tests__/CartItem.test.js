import React from 'react'
import renderer from 'react-test-renderer'
import CartItem from '../CartItem'

test('CartItem renders correctly', () => {
  const component = (
    <CartItem
      item={{
        name: 'ball',
        amount: 3,
        price: 2,
      }}
    />
  )
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
