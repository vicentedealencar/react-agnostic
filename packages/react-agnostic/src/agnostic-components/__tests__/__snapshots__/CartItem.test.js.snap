// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly 1`] = `
<div>
  <span>
    3
  </span>
  <span>
    ball
     - 
    6
  </span>
</div>
`;
